src/web/nextui/src/app/eval/ResultsCharts.tsx

```diff
@@ -415,13 +419,238 @@ function MetricChart({ table }: ChartProps) {
+            type: 'linear',
+            position: 'bottom',
+            ticks: {
+              callback: function (value) {
+                if (Number.isInteger(value)) {
+                  return ordinalSuffixOf(Number(value));
+                }
+                return '';
+              },
+            },
+          },
+          y: {
+            title: {
+              display: true,
+              text: `Pass Rate`,
+            },
+            ticks: {
+              callback: function (value: string | number, index: number, values: any[]) {
+                let ret = String(Math.round(Number(value)));
+                if (index === values.length - 1) {
+                  ret += '%';
+                }
+                return ret;
+              },
+            },
+          },
+        },
+
+        plugins: {
+          legend: {
+            display: true,
+          },
+          tooltip: {
+            callbacks: {
+              title: function (context) {
+                return `Pass Rate: ${context[0].parsed.y.toFixed(2)}%`;
+              },
+              label: function (context) {
+                const point = context.raw as Point;
+                let label = point.metadata.label;
+                if (label && label.length > 30) {
+                  label = label.substring(0, 30) + '...';
+                }
+                return [
+                  `evalId: ${point.metadata.evalId}`,
+                  `Prompt: ${label}`,
+                  `Date: ${point.metadata.date}`,
+                  `Score: ${point.metadata.score}`,
+                ];
+              },
+            },
+          },
+        },
+
+        onClick: function (event, elements) {
+          if (elements.length > 0) {
+            const topMostElement = elements[0];
+            const pointData = (topMostElement.element as any).$context.raw as Point;
+            const evalId = pointData.metadata.evalId;
+            window.open(`/eval/?evalId=${evalId}`, '_blank');
+          }
+        },
+      },
+    });
+  }, [table, evalId, config, datasetId, dataset]);
+
+  return <canvas ref={lineCanvasRef} style={{ maxHeight: '300px', cursor: 'pointer' }}></canvas>;
+}
+
 function ResultsCharts({ columnVisibility }: ResultsChartsProps) {
   const theme = useTheme();
   Chart.defaults.color = theme.palette.mode === 'dark' ? '#aaa' : '#666';
   const [showCharts, setShowCharts] = useState(true);
+  const [showLineChart, setShowLineChart] = useState(false);
+
+  const { table, evalId, config } = useStore();
+
+  useEffect(() => {
+    const fetchDataset = async () => {
+      if (!config) return;
+      const datasetId = createHash('sha256').update(JSON.stringify(config.tests)).digest('hex');
+      fetch(`${await getApiBaseUrl()}/api/evals/${datasetId}`)
+        .then((response) => response.json())
+        .then((data: { data: TestCasesWithMetadata[] }) => {
+          setShowLineChart(data.data[0].count > 1);
+        });
+    };
+    fetchDataset();
```

Comment:
Evaluate the config here i.e.`if(!!config) fetchDataset();`

Reply:
got it, fixed!

Reply:
I see that you're fetching the dataset twice and already have `datasetId` memoized: try moving the fetch call into a separate function which accepts the `datasetId: string` argument.  Something like:

```tsx
const fetchDataset = async (id: string) => {
  try {
   const res = await fetch(`${await getApiBaseUrl()}/api/evals/${id}`);
   return response.json();
  } catch (error) {
    // handle the error!
  }
}

const showLineChart = useMemo(async () => {
  if (datasetId) {
    const data = await fetchDataset(datasetId);
    setShowLineChart(data.data[0].count > 1);
  } else return false;
}, [datasetId])
```

Reply:
Thank you for your feedback! Just update the code with the following changes:
Extract the fetchDataset() function and call it separately in LineChart and ResultChart components. 
Use memo to store the datasetId in ResultChart and pass it down to LineChart to avoid redundant use.